docs/architecture/adr-082-data-companion-api.md

```diff
@@ -0,0 +1,401 @@
+# ADR 082: Data Companion API
+
+## Changelog
+
+- 2022-09-10: First draft (@thanethomson)
+
+## Status
+
+Accepted | Rejected | Deprecated | Superseded by
+
+## Context
+
+This ADR proposes to effectively offload some data storage responsibilities and
+functionality to **single external companion service** in order to:
+
+1. Improve core consensus stability.
+2. Eventually reduce the amount of functionality for which the core team is
+   responsible, thereby improving maintainability.
+3. Still cater for certain use cases that require this data and functionality.
+
+The way the system is currently built is such that a Tendermint node is mostly
+self-contained. While this philosophy initially allowed a certain degree of ease
+of node operation (i.e. simple UX), it has also lent itself to feature sprawl,
+with Tendermint being asked to take care of increasingly more than just
+consensus. Under the spotlight in this ADR are:
+
+1. **Event indexing**, which is required in order to facilitate arbitrary block
+   and transaction querying, as well as subscription for arbitrary events. We
```

Comment:
Why is it necessary to index events in order to facilitate querying blocks and transactions? Events are not represented in either of those entities.

Reply:
The `/tx`, `/tx_search` and `/block_search` endpoints, for instance, all require the indexers to be able to work.

See https://github.com/tendermint/tendermint/blob/9d01a6880e38b79ec31849e436677c70ccd0e588/rpc/core/tx.go#L20

https://github.com/tendermint/tendermint/blob/9d01a6880e38b79ec31849e436677c70ccd0e588/rpc/core/tx.go#L54

https://github.com/tendermint/tendermint/blob/9d01a6880e38b79ec31849e436677c70ccd0e588/rpc/core/blocks.go#L197

Other endpoints, like getting the block by its associated height, do not require the indexer.

Reply:
I understood that indexing of transactions is different to indexing of events. Do I misunderstand the context?

Reply:
Sorry, yes, not sure why I said the `/tx` endpoint too. I should've just posted the `/tx_search` and `/block_search` endpoints. Both of those require events to work.